addons/debugger/timing/createTimingTab.js

```diff
@@ -0,0 +1,154 @@
+import TimingManager from "./TimingManager.js";
+import HeatmapManager from "./HeatmapManager.js";
+import Profiler from "./Profiler.js";
+import { createTableHeader } from "./ui-components/tableHeader.js";
+import TableRows from "./ui-components/TableRows.js"; // Importing the extended LogView class
+import downloadBlob from "../../../libraries/common/cs/download-blob.js";
+import { isPaused, onPauseChanged, getRunningThread } from "../module.js";
+
+export default async function createTimingTab({ debug, addon, console, msg }) {
+  function createContent() {
+    const content = Object.assign(document.createElement("div"), {
+      className: "sa-timing-content",
+    });
+    content.appendChild(tableHeader);
+    content.appendChild(tableRows.outerElement);
+
+    return content;
+  }
+
+  function createLineByLineButton() {
+    const lineByLineButton = debug.createIconButton({
+      text: msg("timing-view-line-by-line"),
+      icon: addon.self.dir + "/icons/tools2.svg",
+    });
+
+    lineByLineButton.element.addEventListener("click", () => {
+      config.showLineByLine = !config.showLineByLine;
+      if (config.showLineByLine && !config.isStepThreadPolluted) {
+        polluteStepThread();
+      } else if (!config.showLineByLine && config.isStepThreadPolluted) {
+        unpollutStepThread();
+      }
+      // Update button text
+      lineByLineButton.element.textContent = config.showLineByLine
+        ? msg("timing-view-timers")
+        : msg("timing-view-line-by-line");
+    });
+
+    return lineByLineButton;
+  }
+
+  function createHeatmapButton() {
+    const heatmapButton = debug.createIconButton({
+      text: msg("timing-show-heatmap"),
+      icon: addon.self.dir + "/icons/tools.svg",
+    });
+
+    heatmapButton.element.addEventListener("click", () => {
+      config.showHeatmap = !config.showHeatmap;
+      if (config.showHeatmap) {
+        heatmapManager.showHeatmapFn(1.0);
+      } else {
+        heatmapManager.hideHeatmapFn();
+      }
+      // Update button text
+      heatmapButton.element.textContent = config.showHeatmap ? msg("timing-hide-heatmap") : msg("timing-show-heatmap");
+    });
+
+    return heatmapButton;
+  }
+
+  function createExportButton() {
+    const exportButton = debug.createIconButton({
+      text: msg("export"),
+      icon: addon.self.dir + "/icons/download-white.svg",
+    });
+
+    exportButton.element.addEventListener("click", () => {
+      const csvContent = tableRows.rows.map((row) => Object.values(tableRows.getRowValues(row)).join(",")).join("\n");
```

Comment:
In line-by-line mode the first column is mostly block IDs inclusing the filename, so that should either be fixed or the button disabled in that mode.